Move text-direction discussion from Wiki to new /docs area (#20)
w3c/rdf-star-wg: docs/text-direction.md

According to the [W3C Process Document](https://www.w3.org/2021/Process-20211102/):

> All reports, publications, or other deliverables produced by the group for public consumption should follow best practices for **internationalization** and for accessibility to people with disabilities. Network access to W3C-controlled domains may be assumed.

> One thing that wasn't given as much attention in earlier Working Groups (including RDF 1.1) was support for describing the base paragraph direction that text should should use for presentation. Although Unicode provides some information about the directionality of individual code points and a way to estimate base paragraph direction when no other information is available (see [First-string property detection](https://w3c.github.io/string-meta/#firststrong), there are still many cases in which this results in incorrect results. The requirements here are captured  in https://github.com/w3c/rdf-concepts/issues/9 on RDF Concepts, where much of this information has been gathered. Of course, changes potentially will impact other specifications.

Reviewer: ```suggestion
One thing that wasn't given as much attention in earlier Working Groups (including RDF 1.1) was support for describing the base paragraph direction that text should should use for presentation. Although Unicode provides some information about the directionality of individual code points and a way to estimate base paragraph direction when no other information is available (see [First-strong property detection](https://w3c.github.io/string-meta/#firststrong), there are still many cases in which this results in incorrect results. The requirements here are captured  in https://github.com/w3c/rdf-concepts/issues/9 on RDF Concepts, where much of this information has been gathered. Of course, changes potentially will impact other specifications.
```

> One thing that wasn't given as much attention in earlier Working Groups (including RDF 1.1) was support for describing the base paragraph direction that text should should use for presentation. Although Unicode provides some information about the directionality of individual code points and a way to estimate base paragraph direction when no other information is available (see [First-strong property detection](https://w3c.github.io/string-meta/#firststrong), there are still many cases in which this results in incorrect results. The requirements here are captured  in https://github.com/w3c/rdf-concepts/issues/9 on RDF Concepts, where much of this information has been gathered. Of course, changes potentially will impact other specifications.

Reviewer 2: ```suggestion
One thing that wasn't given as much attention in earlier Working Groups (including RDF 1.1) was support for describing the base paragraph direction that text should should use for presentation. Although Unicode provides some information about the directionality of individual code points and a way to estimate base paragraph direction when no other information is available (see [First-strong property detection](https://w3c.github.io/string-meta/#firststrong), there are still many cases in which this delivers incorrect results. The requirements here are captured in [issue #9 on RDF Concepts](https://github.com/w3c/rdf-concepts/issues/9), where much of this information has been gathered. Of course, changes will potentially impact other specifications.
```

> The [Internationalization Best Practices for Spec Developers Note](https://www.w3.org/TR/international-specs/) has a whole section devoted to [Text direction](https://www.w3.org/TR/international-specs/#text_direction), and this is a checklist item for the internationalization review all specs need to go through. When this was performed for JSON-LD 1.1, it came up that the underlying data model (RDF) had no way to indicate text direction, which was a shortcoming. There were a number of review items and proposals put forth (See [RDF Literals and Base Directions](https://w3c.github.io/rdf-dir-literal/) by Ivan Herman and Pierre-Antoine Champin, for example and further discussion).

Reviewer 2: ```suggestion
The [_Internationalization Best Practices for Spec Developers_ Note](https://www.w3.org/TR/international-specs/) has a whole section devoted to [text direction](https://www.w3.org/TR/international-specs/#text_direction), and this is a checklist item for the internationalization review all specs need to go through. When this was performed for JSON-LD 1.1, it was noted that the underlying data model (RDF) had no way to indicate text direction, which was a shortcoming. There were a number of review items and proposals put forth (see [_RDF Literals and Base Directions_](https://w3c.github.io/rdf-dir-literal/) by Ivan Herman and Pierre-Antoine Champin, for example and further discussion).
```

> JSON-LD added support by introducing the [`@direction` keyword](https://www.w3.org/TR/json-ld11/#dfn-base-direction) added to a Literal Object. When serializing to and from an RDF Dataset, non-normative solutions were proposed using [The `i18n` Namespace](https://www.w3.org/TR/json-ld11/#the-i18n-namespace) and [The `rdf:CompoundLiteral`](https://www.w3.org/TR/json-ld11/#the-rdf-compoundliteral-class-and-the-rdf-language-and-rdf-direction-properties) along with associated properties. The idea being that these would be interim solutions available before normative behavior could be described in a future version of RDF.

Reviewer 2: ```suggestion
JSON-LD added support by introducing the [`@direction` keyword](https://www.w3.org/TR/json-ld11/#dfn-base-direction) added to a Literal Object. When serializing to and from an RDF Dataset, non-normative solutions were proposed using [the `i18n` Namespace](https://www.w3.org/TR/json-ld11/#the-i18n-namespace) and [the `rdf:CompoundLiteral`](https://www.w3.org/TR/json-ld11/#the-rdf-compoundliteral-class-and-the-rdf-language-and-rdf-direction-properties) along with associated properties. The idea was that these interim solutions would be available before normative behavior could be described in a future version of RDF.
```

Also see the [Pros and Cons](https://w3c.github.io/rdf-dir-literal/#pros-and-cons-0) discussion in [RDF Literals and Base Directions](https://w3c.github.io/rdf-dir-literal/),

## The `rdf:CompoundLiteral` class

Summary: The `[i18n` namespace](rdf:CompoundLiteral) uses a blank node to contain the separate components of the "literal":

Example for Arabic text (`ar-EG`) with `right to left` text direction (`rtl`):

```turtle
[
  ex:title [
    rdf:value "HTML و CSS: تصميم و إنشاء مواقع الويب",
    rdf:language "ar-eg",
    rdf:direction "rtl"
  ];
  ex:publisher [
    rdf:value "مكتبة",
    rdf:language "ar-eg",
    rdf:direction "rtl"
  ]
] .
```

**Ednote**: The group considered using a language-tagged string as the value of `rdf:value`, but I can't immediately see the rationale for rejecting this.

Advantages:
* Works in every existing RDF serialization format.

Disadvantages:
* Introduces a new blank node.
* Also same dis-advantages as the `i18n` datatype solution.

Also see the [Pros and Cons](https://w3c.github.io/rdf-dir-literal/#pros-and-cons-2) discussion in [RDF Literals and Base Directions](https://w3c.github.io/rdf-dir-literal/),

## Extend `langString`

> As expressed in [RDF Literals and Base Directions](https://w3c.github.io/rdf-dir-literal/) [Extend `langString`](https://w3c.github.io/rdf-dir-literal/#extending-lang-string), the definition for `rdf:langString` could be extended to include _both_ the language tag and the text direction by introducing a separator such as a carat (`^`).

Reviewer 2: ```suggestion
As expressed in [RDF Literals and Base Directions](https://w3c.github.io/rdf-dir-literal/), the [definition for `rdf:langString` could be extended](https://w3c.github.io/rdf-dir-literal/#extending-lang-string) to include _both_ the language tag and the text direction by introducing a separator such as a carat (`^`).
```
